aggregator/src/aggregator/aggregation_job_continue.rs

```diff
@@ -230,14 +230,30 @@ impl VdafOps {
             })
             .with_last_continue_request_hash(request_hash);
 
+        // Write accumulated aggregation values back to the datastore; mark any reports that can't
+        // be aggregated because the batch is collected with error BatchCollected.
+        let unwritable_reports = accumulator.flush_to_datastore(tx, &vdaf).await?;
+        for report_aggregation in &mut report_aggregations {
+            if unwritable_reports.contains(report_aggregation.report_id()) {
+                *report_aggregation = report_aggregation
+                    .clone()
+                    .with_state(ReportAggregationState::Failed(
+                        ReportShareError::BatchCollected,
+                    ))
+                    .with_last_prep_step(Some(PrepareStep::new(
+                        *report_aggregation.report_id(),
+                        PrepareStepResult::Failed(ReportShareError::BatchCollected),
+                    )));
+            }
+        }
```

Comment:
What if _all_ the reports in the aggregation job are now failed with `BatchCollected`. Should the state of the aggregation job go to `AggregationJobState::Finished`?